packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -222,10 +222,17 @@ export class TableLayoutManager<TData extends TableRecord> {
     }
 
     private resetGridSizedColumns(): void {
-        if (!this.gridSizedColumns) {
+        if (!this.gridSizedColumns || this.gridSizedColumns.length === 0) {
```

Comment:
If you want, you could simplify this to one check. I don't have a strong preference either way.

```suggestion
        if (!this.gridSizedColumns?.length) {
```